presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSink.java

```diff
@@ -222,7 +229,7 @@ public void appendPage(Page page)
 
             for (int position = 0; position < page.getPositionCount(); position++) {
                 int bucket = bucketFunction.isPresent() ? bucketFunction.get().getBucket(bucketArgs, position) : 0;
-                int day = temporalFunction.isPresent() ? temporalFunction.get().getDay(temporalBlock, position) : 0;
+                int day = temporalFunction.isPresent() ? temporalFunction.get().getDay(session, temporalBlock, position) : 0;
```

Comment:
Session was not necessary before, why is it needed now?

Reply:
This allows us to use getObject instead of assuming Timestamp format is UTC offset, so perhaps easier to expand later for new format?

Reply:
We don't need session here and should not be using `getObjectValue()` as that is extremely inefficient and is only used to return values to the client. We are not "assuming" the timestamp representation: it is well-defined for the type.